### PR TITLE
OBPIH-5229 allow filter by payment terms in po list page

### PR DESCRIPTION
**Pull request by drodzewicz**

### CODE REVIEW

**Review by awalkowiak — APPROVED** [review]

File: src/js/actions/index.js [hunk 1]
```diff
@@ -424,6 +426,17 @@ export function fetchPurchaseOrderStatuses() {
   };
 }
 
+export function fetchPaymentTerms() {
+  return (dispatch) => {
+    purchaseOrderApi.getPaymentTerms().then((res) => {
```
Comment (kchelstowski) [inline src/js/actions/index.js:431]: for this simple case it doesn't matter, but IMO async/await convention is generally cleaner

File: src/js/hooks/list-pages/purchase-order/usePurchaseOrderListTableData.jsx [hunk 2]
```diff
@@ -27,14 +27,15 @@ const usePurchaseOrderListTableData = (filterParams) => {
...
-      status, origin, orderedBy, createdBy, destinationParty, destination,
+      status, origin, orderedBy, createdBy, destinationParty, destination, paymentTerm,
     } = filterParams;
     return {
       ..._.omitBy({
         offset: `${offset}`,
         max: `${state.pageSize}`,
         ...sortingParams,
         ...filterParams,
+        paymentTerm: paymentTerm && paymentTerm.map(term => term.id),
```
Comment (kchelstowski) [inline src/js/hooks/list-pages/purchase-order/usePurchaseOrderListTableData.jsx:38]: could be simplified to:
```javascript
paymentTerm: paymentTerm?.map?.(({ id }) => id),
```

File: src/js/reducers/purchaseOrderReducer.jsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { FETCH_PURCHASE_ORDER_STATUSES } from 'actions/types';
+import { FETCH_PAYMENT_TERMS, FETCH_PURCHASE_ORDER_STATUSES } from 'actions/types';
 
 const initialState = {
   statuses: [],
```
Comment (kchelstowski) [inline src/js/reducers/purchaseOrderReducer.jsx:4]: you haven't added an `initialState` for `paymentTerms`. I hope it won't break anything and won't cause any null pointers

**Review by kchelstowski — COMMENTED** [review]
"post-factum" review

File: src/js/reducers/purchaseOrderReducer.jsx (hunk 3 above)
Comment (drodzewicz) [reply src/js/reducers/purchaseOrderReducer.jsx:4]: Thanks for spotting that, I'll make sure to add it in one of the other PR's tat I'l be working on

**Pull request merged**